### PR TITLE
Support  JAVA_HOME in env file when check java version

### PR DESCRIPTION
**Pull request by gaozhangmin**
### Motivation
1、Currently, If we set JAVA_HOME in env file, but `IS_JAVA_8 ` always use `java -version` to check java version.
It will get wrong version if we set JAVA_HOME, thus, wrong gc parameters are set.

### Modifications
Check if we have set JAVA_HOME or not. before check java version.

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: conf/bkenv.sh [hunk 1]
```diff
@@ -32,15 +32,21 @@ BOOKIE_CONF=${BOOKIE_CONF:-"$BK_HOME/conf/bookkeeper.conf"}
 # BOOKIE_LOG_CONF=
 
 # Logs location
-# BOOKIE_LOG_DIR=
+# PULSAR_LOG_DIR=
```
Comment (codelipenghui) [inline conf/bkenv.sh:35]: Is it a related change?

Others look good to me.

Comment (gaozhangmin) [reply conf/bkenv.sh:35]: not related @codelipenghui。 Should I open an new pr?

File: conf/bkenv.sh [hunk 2]
```diff
@@ -32,15 +32,21 @@ BOOKIE_CONF=${BOOKIE_CONF:-"$BK_HOME/conf/bookkeeper.conf"}
...
 
 # Memory size options
 BOOKIE_MEM=${BOOKIE_MEM:-${PULSAR_MEM:-"-Xms2g -Xmx2g -XX:MaxDirectMemorySize=2g"}}
 
 # Garbage collection options
 BOOKIE_GC=${BOOKIE_GC:-${PULSAR_GC:-"-XX:+UseG1GC -XX:MaxGCPauseMillis=10 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC"}}
 
-IS_JAVA_8=`java -version 2>&1 |grep version|grep '"1\.8'`
+if test -z "$JAVA_HOME"
```
Comment (lhotari) [inline conf/bkenv.sh:43]: replace `test` with `[` ... `]`
```suggestion
if [ -z "$JAVA_HOME" ]
```

File: conf/pulsar_env.sh [hunk 3]
```diff
@@ -48,7 +48,12 @@ PULSAR_MEM=${PULSAR_MEM:-"-Xms2g -Xmx2g -XX:MaxDirectMemorySize=4g"}
 PULSAR_GC=${PULSAR_GC:-"-XX:+UseG1GC -XX:MaxGCPauseMillis=10 -XX:+ParallelRefProcEnabled -XX:+UnlockExperimentalVMOptions -XX:+DoEscapeAnalysis -XX:ParallelGCThreads=32 -XX:ConcGCThreads=32 -XX:G1NewSizePercent=50 -XX:+DisableExplicitGC"}
 
 # Garbage collection log.
-IS_JAVA_8=`java -version 2>&1 |grep version|grep '"1\.8'`
+if test -z "$JAVA_HOME"
```
Comment (lhotari) [inline conf/pulsar_env.sh:51]: replace `test` with `[` ... `]`
```suggestion
if [ -z "$JAVA_HOME" ]
```

**Review by lhotari — CHANGES_REQUESTED** [review]

File: conf/bkenv.sh (hunk 1 above)
Comment (lhotari) [reply conf/bkenv.sh:35]: yes, would be better to revert the change since it doesn't look related to JAVA_HOME changes

**Review by codelipenghui — APPROVED** [review]

**Review by lhotari — APPROVED** [review]

**Pull request merged**